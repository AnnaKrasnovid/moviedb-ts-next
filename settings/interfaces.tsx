export interface GenreItemInt {
    name: string,
}
export interface MovieInt {
    id: number,
    name: string,
    type: string,
    alternativeName: string,
    year: number,
    description: string,
    shortDescription: string,
    rating: {
        kp: number,
        imdb: number,
    },
    videos: {
        trailers: {
            url: string,
        }
    },
    persons: Array<
        {
            id: number,
            photo: Array<string>,
            profession: string,
            name: string,
        }
    >,
    movieLength: number,
    poster: {
        url: string,
    }
    countries: Array<{ name: string, }>
    genres: Array<{ name: string, }>
}

type ButtonTypes = 'button' | 'submit';
export interface ButtonInt {
    type?: ButtonTypes,
    callback: () => void
}

export interface ButtonTextInt extends ButtonInt {
    text?: string,
}

export interface SubmenuItemInt {
    id: string,
    path: string,
    title: string,
    value: string,
}

export interface MenuItemInt {
    id: string,
    path: string,
    title: string,
    submenu?: Array<SubmenuItemInt>
}
