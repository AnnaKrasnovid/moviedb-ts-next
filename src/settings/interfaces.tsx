import { ReactNode, Dispatch, SetStateAction } from "react";
export interface GenreItemInt {
    name: string,
}

type ButtonTypes = 'button' | 'submit';
export interface ButtonInt {
    type?: ButtonTypes,
    callback: () => void,
    className?: string,
}

export interface ButtonCloseInt extends ButtonInt {
    position?: 'left' | 'right',
}

export interface ButtonTextInt extends ButtonInt {
    text?: string,
}

export interface SubmenuItemInt {
    id: string,
    path: string,
    title: string,
    value: string,
}

export interface MenuItemInt {
    id: string,
    path: string,
    title: string,
    submenu?: Array<SubmenuItemInt>
}

export interface ValueInt {
    value: string,
}

export interface NameInt {
    name: string,
}

export interface MovieBaseInt {
    alternativeName: string | null,
    countries: Array<{ name: string }>,
    description: string | null,
    enName: string | null,
    externalId: {
        imdb: string | any,//null
        kpHD: string | any,//null
    },
    genres: Array<{ name: string }>,
    id: number,
    logo?: {
        url: string | null,
    },
    movieLength: number | null,
    name: string,
    names: Array<{ name: string }>,
    poster: {
        previewUrl: string | null,
        url: string | null,
    },
    rating: {
        await: number,
        filmCritics: number,
        imdb: number,
        kp: number,
        russianFilmCritics: number,
    },
    shortDescription: string | null,
    type: string,
    votes: {
        kp: number | null,
        imdb: number | null,
        filmCritics: number | null,
        russianFilmCritics: number | null,
        await: number | null
    },
    watchability: any,
    year: number,
    releaseYears?:
    { start: number, end: number }
}

export interface MovieSimpleInt {
    alternativeName?: string | null,
    description: string | null,
    enProfession?: string | null,
    general: boolean,
    id: number,
    name: string | null,
    rating: number | null
}

export interface MovieSemilarItemInt {
    alternativeName: string | null,
    enName: string | null,
    id: number,
    name: string | null,
    poster: {
        previewUrl: string | null,
        url: string | null,
    },
    type: string
}

export interface SpousesInt {
    id: number,
    name: string | null,
    divorced: boolean,
    divorcedReason: string,
    sex: string,
    children: number,
    relation: string
}

export interface ActorItemInt {
    age: number,
    birthPlace: Array<ValueInt>,
    birthday: string,
    countAwards: number | null,
    death: string | null,
    deathPlace: string | null,
    enName: string | null,
    facts: Array<ValueInt>,
    growth: number | null,
    id: number,
    isParse: boolean,
    movies: Array<MovieSimpleInt>
    name: string,
    photo: string | null,
    profession: Array<ValueInt>,
    sex: number,
    spouses: Array<SpousesInt> | null,
    updatedAt: string,
}

export interface ActorSimpleItemInt {
    description: string,
    enName: string | null,
    enProfession: string | null,
    id: number,
    name: string,
    photo: string,
    profession: string | null,
}
export interface ActorsListInt {
    list: Array<ActorSimpleItemInt>,
}
export interface ActorInt {
    actor: ActorItemInt,
    error?: string
}

export interface ActorCardInt {
    item: ActorSimpleItemInt
}

export interface FactsInt {
    list: Array<ValueInt>
}

export interface MovieItemInt extends MovieBaseInt {
    ageRating: number | null,
    backdrop: { previewUrl: string | null, url: string | null },
    budget: {
        value: number,
        currency: string
    },
    deletedAt: any, // null
    distributors: {
        distributor: any,//null
        distributorRelease: any,// null
    }
    facts: Array<{ value: string }>,
    fees: {
        russia: any,
        usa: any,
        world: any,
    },
    images: { framesCount: number },
    imagesInfo: { framesCount: number }
    isSeries: boolean,
    lists: Array<any>,
    persons: Array<ActorSimpleItemInt>,
    premiere: any,
    productionCompanies: Array<any>,
    ratingMpaa: number | null,
    seasonsInfo: Array<any>,
    sequelsAndPrequels: Array<any>
    seriesLength: number | null,
    similarMovies: Array<MovieSemilarItemInt>,
    slogan: string,
    spokenLanguages: {
        name: string | null,
        nameEn: string | null
    },
    status: string,
    technology: { hasImax: boolean, has3D: boolean },
    ticketsOnSale: boolean,
    top10: number | null,
    top250: number | null,
    totalSeriesLength: number | null,
    typeNumber: number,
    updateDates: Array<string>,
    updatedAt: string,
    videos: {
        trailers: Array<{
            name: string | null,
            site: string | null,
            type: string | null,
            url: string | null,
        }>,
        teasers: Array<any>
    },
}

export interface MovieCardSimpleInt {
    item: MovieSemilarItemInt
}

export interface MovieCardInt {
    movie: MovieItemInt
}

export interface SimilarMovieCardInt {
    alternativeName: string,
    enName: string | null,
    id: number,
    name: string,
    poster: {
        previewUrl: string,
        url: string
    },
    type: string
}

export interface SimilarMoviesInt {
    list: Array<SimilarMovieCardInt>
}

export interface MoviesInfoInt {
    docs: Array<MovieBaseInt>,
    limit: number,
    page: number,
    pages: number,
    total: number,
}

export interface MoviesPageInt {
    movies: MoviesInfoInt,
    error?: string
}

export interface MainPageInt {
    movieRating: MoviesInfoInt,
    cartoons: MoviesInfoInt,
    series: MoviesInfoInt,
    movieRandom: MovieItemInt,
    error?: Array<string>
}

export interface CarouselInt {
    children: ReactNode,
    slides?: [number, number, number, number, number] | number,
    className?: string,
    loop?: boolean,
    autoplay?: boolean,
    navigation?: boolean
}

export interface DescriptionMovieInt {
    title: string,
    info: string | number,
}

export interface DescriptionMovieItemInt {
    title: string,
    info?: string | number,
}

export interface FilterInt {
    id: string,
    title: string,
    value: string
}

export interface GenreCardInt {
    index: number,
    item: SubmenuItemInt,
}

export interface GridMoviesInt {
    children: ReactNode
}

export interface HeaderInt {
    onOpenMenu: () => void,
    onClosePopup: () => void
}

export interface InfiniteScrollInt {
    children: ReactNode,
    callback: () => void,
    condition: boolean, // условие при котором не будет вызываться функция
}

type LogoTypes = 'header' | 'footer' | 'menu';
export interface LogoInt {
    type: LogoTypes,
}

export interface MenuMobileInt {
    onClosePopup: () => void,
    isOpenPopupMenu: boolean,
}

type MovieTypesCard = 'small' | 'middle';

export interface MovieCardMainInt {
    item: MovieBaseInt,
    type?: MovieTypesCard
}

export interface MoviesListInt {
    list: Array<MovieBaseInt>,
    pages?: number | undefined,
    error?: string
}

type NavigationTypes = 'header' | 'menu';

export interface NavigationInt {
    type: NavigationTypes,
    onClosePopup: () => void
}

export interface NavigationLinkInt {
    item: MenuItemInt,
    isActiveSubmenu: boolean,
    closePopup: () => void,
}

type RatingRoundTypes = 'yellow' | 'orange';

export interface RatingRoundInt {
    number: number,
    type?: RatingRoundTypes,
}

export interface PersonPageInt {
    actor: ActorItemInt,
}

type SocialButtonsTypes = 'footer' | 'menu';
export interface SocialButtonsInt {
    type: SocialButtonsTypes
}

export interface SubmenuInt {
    item: MenuItemInt,
    isActiveSubmenu: boolean,
}

export interface CompilationInt {
    title: string,
    moviesList: Array<MovieBaseInt>,
    link: string
}

export interface MoviePageInt {
    movie: MovieItemInt,
    error?: string
}

type ArrowTypes = 'select' | 'menu';
export interface ArrowIconInt {
    isActive: boolean,
    type?: ArrowTypes,
}

export interface InputSearchInt {
    searchValue: string,
    setSearchValue: Dispatch<SetStateAction<string>>
}

export interface ScrollBarInt {
    children: ReactNode,
}

export interface OverlayInt {
    children: ReactNode
    isOpenPopup: boolean
    closePopup?: () => void
    className?: string
}

export interface OptionItemInt {
    id: string,
    title: string,
    value: string,
}

export interface SelectInt {
    options: Array<OptionItemInt>,
    label?: string,
    placeholder?: string,
    callback: (value: string) => void,
    defaultValue?: any,
}

export interface LayoutInt {
    children: ReactNode,
}

export interface DescriptionMovieCardInt {
    movie: MovieItemInt
}

export interface OptionInt {
    hour12?: boolean,
    weekday?: "long" | "short" | "narrow" | undefined,
    year?: "numeric" | "2-digit" | undefined,
    month?: "long" | "short" | "narrow" | "numeric" | "2-digit" | undefined,
    day?: "numeric" | "2-digit" | undefined,
    hour?: "numeric" | "2-digit" | undefined,
    minute?: "numeric" | "2-digit" | undefined,
    second?: "numeric" | "2-digit" | undefined,
    timeZoneName?: "long" | "short" | "shortOffset" | "longOffset" | "shortGeneric" | "longGeneric" | undefined,
}

export interface TooltipInt {
    isOpenTooltip: boolean,
    text?: string| ReactNode,
}

export interface NotFoundInt {
    status?: string,
    message?: string
}

export interface InformationInt {
    text: string | ReactNode
}