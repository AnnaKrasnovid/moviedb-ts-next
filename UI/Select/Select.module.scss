@import '../../styles/variables.scss';

.dropdown-select {
    // max-width: 300px;
    width: 100%;
    border-radius: 5px;
    display: flex;
    flex-direction: column;
    position: relative;
    // border: 1px solid var(--border-color);
    margin-top: 5px;
    height: 38px;

    &_active {
        box-shadow: 0px 4px 8px rgba(0, 0, 0, 0.05);
        // border: 1px solid var(--border-color);
        border-radius: 5px 5px 0 0;
        // background-color: var(--background-color);
    }

    &__levels {  
        box-shadow: 0px 4px 8px rgba(0, 0, 0, 0.05);
        width: calc(100% + 2px);
        position: absolute;
        top: 36px;
        left: -1px;
        z-index: 1;
        // background-color: var(--background-color);
        // border: 1px solid var(--border-color);
        border-radius: 0 0 5px 5px;
        border-top: 0;
        max-height: 300px;
        visibility: hidden;
        opacity: 0;
        transition: visibility .2s linear, opacity .2s linear;
        overflow: hidden;

        background: rgba(28, 29, 31, 0.5);
    border: 1px solid #2B2B2B;
    box-shadow: 0px 7px 16px rgba(0, 0, 0, 0.25);
    -webkit-backdrop-filter: blur(20px);
    backdrop-filter: blur(20px);
    border-radius: 8px;
    margin-top: 2px;

        &_active {            
            border: 1px solid #424242;
            // border-top: 1px solid $color-light-green;
            visibility: visible;
            opacity: 1;
            transition: visibility .2s linear, opacity .2s linear;
        }
    }

    &__button {
        width: 100%;
        border: 0;
        padding: 0;
        border-radius: 5px;
        display: grid;
        grid-template-columns: 1fr 7px;
        column-gap: 8px;
        align-items: center;
        padding: 0 25px;        
        height: 36px;
        color: #fff;      
        border: 1px solid #424242;
        background-color: #1B1E1E;
        border-radius: 8px;
        box-shadow: 0px 7px 16px rgba(0, 0, 0, 0.2);
        backdrop-filter: blur(20px);
        cursor: pointer;
    }

    &__title {
        font-weight: 400;
        font-size: 14px;
        line-height: 14px;
        color: var(--color);
        justify-self: start;
    }

    &__list {
        margin: 0;
        padding: 0;
        display: grid;
        grid-template-rows: 1fr;
        row-gap: 8px;
        padding: 8px 0;
         max-height: 300px;
    }

    &__item {
        padding: 0 18px;
        display: flex;
        align-items: center;
        justify-content: space-between;
        min-height: 30px;
cursor: pointer;
        &:hover {
            background: rgba(28, 29, 31, 0.5);
            color: $color-primary;
        }

        &_type {
            &_multi-select {
                display: grid;
                grid-template-columns: 1fr 10px;
                column-gap: 8px;
            }
        }

        &_active {
            background: rgba(28, 29, 31, 0.5);
            color: $color-primary;
        }
    }

    &__point {
        width: 7px;
        height: 7px;
        box-shadow: 0px 0px 4px rgba(122, 204, 231, 0.6);
        // background: $color-middle-gray;
        display: inline-block;
        border-radius: 50%;
    }

    &__done {
        // display: none;
        // color: $color-light-gray;

        &_active {
            display: block;
            // color: $color-middle-green;
        }
    }

    // .scroll {
    //     padding: 0;
    // }
}

.label-input {
    // color: var(--color);
}

// .scroll_type_select {
//     padding: 5px 0;
// }