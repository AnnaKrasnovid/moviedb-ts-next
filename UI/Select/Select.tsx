import { useEffect, useState, SyntheticEvent } from 'react';

import ScrollBar from '../ScrollBar/ScrollBar';
import ArrowIcon from '../ArrowIcon/ArrowIcon';
// import { SubmenuItemInt } from '../../settings/interfaces';

import styles from './Select.module.scss';

export interface OptionItemInt {
    id: string,
    title: string,
    value: string,
}

export interface SelectInt {
    options: Array<OptionItemInt>,
    label?: string,
    placeholder?: string,
    callback: (value: string) => void,
    defaultValue?: any, 
}

function Select({ options, label, placeholder = 'Выберите...', callback, defaultValue }: SelectInt) {
    const [isActiveDropdown, setIsActiveDropdown] = useState(false);
    const [selectedItem, setSelectedItem] = useState<string>(placeholder);

    const openDropdown = () => {
        setIsActiveDropdown(true);
    }

    const closeDropdown = () => {
        setIsActiveDropdown(false);
    }

    const toggleDropdown = (e: SyntheticEvent) => {
        if (isActiveDropdown) {
            closeDropdown();
        } else {
            e.stopPropagation();
            openDropdown();
        }
    }

    const choiceOption = (id: string | number) => {
        const item = options.find((i) => i.id === id)
        if (item) {
            setSelectedItem(item.title);
            callback(item.value);
        }
    }   

    useEffect(() => {      
       const value= defaultValue === options[0].title ? options[0].title : defaultValue;   
       // @ts-ignore
       setSelectedItem(value);
    }, [defaultValue])

    useEffect(() => {
        document.addEventListener('click', closeDropdown);
    }, [isActiveDropdown])

    return (
        <div className=''>
            <p className={styles['label-input']}>{label}</p>
            <div className={`${styles['dropdown-select']} ${isActiveDropdown ? styles['dropdown-select_active'] : ''}`} onClick={(e) => toggleDropdown(e)}>
                <div className={styles['dropdown-select__button']}>
                    <span className={`${styles['dropdown-select__title']} ${styles['dropdown-select__title-button']}`}>{selectedItem}</span>
                    <span className={`${styles['arrow']} ${isActiveDropdown ? styles['arrow_up'] : ''}`}>
                        <ArrowIcon isActive={isActiveDropdown} />
                    </span>
                </div>
                <div className={`${styles['dropdown-select__levels']} ${isActiveDropdown ? styles['dropdown-select__levels_active'] : ''}`}>
                    <ScrollBar>
                        <ul className={styles['dropdown-select__list']}>
                            {options.map((item) => (
                                <li className={`${styles['dropdown-select__item']} ${item.title === selectedItem ? styles['dropdown-select__item_active'] : ''}`} key={item.id} onClick={() => choiceOption(item.id)}>
                                    <span className={styles['dropdown-select__title']}>{item.title}</span>
                                </li>
                            ))}
                        </ul>
                    </ScrollBar>
                </div>
            </div>
        </div>
    )
}

export default Select;