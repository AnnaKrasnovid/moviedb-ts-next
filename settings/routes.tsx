export const routes = {
    INDEX: '/',
    MOVIES: '/movies',
    MOVIE: '/movie',
    GENRES: '/genres',
    SERIALS: '/serials',
    CARTOONS: '/cartoons',
    SEARCH: '/search',
};

export const genresRoutes = {
    ACTION_MOVIE: 'action-movie',
    MILITARY: 'military',
    DETECTIVE: 'detective',
    FAMILY: 'family',
    COMEDY: 'comedy',
    CRIMINAL: 'criminal',
    MELODRAMA: 'melodrama',
    THRILLER: 'thriller',
    HORROR: 'horror',
    FANTASTIC: 'fantastic',
    FANTASY: 'fantasy',
    DRAMA: 'drama',
};