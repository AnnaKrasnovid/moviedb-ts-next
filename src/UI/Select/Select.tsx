import { useEffect, useState, memo } from 'react';

import ScrollBar from '@/UI/ScrollBar/ScrollBar';
import OptionsList from './Components/OptionsList/OptionsList';
import SelectButton from './Components/SelectButton/SelectButton';

import { useToggleVisibility } from '../../hooks/useToggleVisibillity';
import { SelectInt } from '@/settings/interfaces';

import styles from './Select.module.scss';

function Select({ options, label = '', callback, defaultValue }: SelectInt) {
    const { ref, isActive, openModal, closeModal } = useToggleVisibility(false);
    const [selectedItem, setSelectedItem] = useState<string>(options[0].title);

    const choiceOption = (id: string | number) => {
        closeModal();
        const item = options.find((i) => i.id === id);

        if (item) {
            setSelectedItem(item.title);
            callback(item.value);
        }
    }

    const getTitle = () => {
        const defaultTitle = options.find((item) => item.value === defaultValue);
        const value = defaultValue && defaultTitle ? defaultTitle.title : options[0].title;
        setSelectedItem(value);
    }

    useEffect(() => {
        getTitle();
    }, [defaultValue])

    return (
        <div className={styles['dropdown']}>
            {label &&
                <p className={styles['dropdown__label']} data-testid='select-label'>
                    {label}
                </p>
            }
            <div className={`${styles['dropdown__select']} ${isActive ? styles['dropdown__select_active'] : ''}`} >
                <SelectButton isActive={isActive} openModal={openModal} selectedItem={selectedItem} />
                <div className={`
                    ${styles['dropdown__options']} 
                    ${isActive ? styles['dropdown__options_active'] : ''}`}
                    ref={ref}
                    data-testid='dropdown-options'
                >
                    <ScrollBar>
                        <OptionsList options={options} choiceOption={choiceOption} selectedItem={selectedItem} />
                    </ScrollBar>
                </div>
            </div>
        </div>
    )
}

export default memo(Select);