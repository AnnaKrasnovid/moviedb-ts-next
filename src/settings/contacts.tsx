export const links = {
    FACEBOOK: '#',
    INSTAGRAM: '#',
    TWITTER: '#',
}