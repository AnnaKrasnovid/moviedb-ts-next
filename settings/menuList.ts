import { genresRoutes, routes } from "./routes";
import { MenuItemInt,SubmenuItemInt } from "./interfaces";

export const submenuGenres:Array<SubmenuItemInt> = [
  { id: '1', path: genresRoutes.ACTION_MOVIE, title: 'Боевики', value: 'боевик' },
  { id: '2', path: genresRoutes.MILITARY, title: 'Военные', value: 'военный' },
  { id: '3', path: genresRoutes.DETECTIVE, title: 'Детективы', value: 'детектив' },
  { id: '4', path: genresRoutes.FAMILY, title: 'Семейные', value: 'семейный' },
  { id: '5', path: genresRoutes.DRAMA, title: 'Драмы', value: 'драма' },
  { id: '6', path: genresRoutes.COMEDY, title: 'Комедии', value: 'комедия' },
  { id: '7', path: genresRoutes.CRIMINAL, title: 'Криминал', value: 'криминал' },
  { id: '8', path: genresRoutes.MELODRAMA, title: 'Мелодрамы', value: 'мелодрама' },
  { id: '9', path: genresRoutes.THRILLER, title: 'Триллеры', value: 'триллер' },
  { id: '10', path: genresRoutes.HORROR, title: 'Ужасы', value: 'ужасы' },
  { id: '11', path: genresRoutes.FANTASTIC, title: 'Фантастика', value: 'фантастика' },
  { id: '12', path: genresRoutes.FANTASY, title: 'Фэнтези', value: 'фэнтези' },
];

export const menu: Array<MenuItemInt> = [
  {
    id: '1',
    path: routes.INDEX,
    title: 'Главная',
  },
  {
    id: '2',
    path: routes.MOVIES,
    title: 'Фильмы',
  },
  {
    id: '3',
    path: routes.GENRES,
    title: 'По жанру',
    submenu: submenuGenres,
  },
  {
    id: '4',
    path: routes.SERIALS,
    title: 'Сериалы',
  },
  {
    id: '5',
    path: routes.CARTOONS,
    title: 'Мультфильмы',
  },
];

