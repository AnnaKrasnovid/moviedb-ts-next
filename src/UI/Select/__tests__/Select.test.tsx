import '@testing-library/jest-dom';
import { render, screen } from "@testing-library/react";
import userEvent from '@testing-library/user-event';

import Select from '../Select';
import { FilterInt } from '@/settings/interfaces';

const onClick = jest.fn();
const mockList: Array<FilterInt> = [
    { id: '0', title: 'Все жанры', value: '' },
    { id: '1', title: 'Боевик', value: 'боевик' },
    { id: '2', title: 'Военный', value: 'военный' },
];

describe('Select', () => {
    it('Рендер компонента с обязательными пропсами', () => {
        render(
            <Select
                options={mockList}
                callback={onClick}
            />
        );

        const label = screen.queryByTestId('select-label');
        const optionDefault = screen.getByTestId('select-button');

        expect(optionDefault).toContainHTML(mockList[0].title);
        expect(label).toBeNull();
    });

    it('Рендер компонента с label', () => {
        render(
            <Select
                options={mockList}
                callback={onClick}
                label={'Название select'}
            />
        );

        const label = screen.queryByText('Название select');
        expect(label).toBeInTheDocument();
    });

    it('Открытие/выбор опции/закрытие dropdown', async () => {
        render(
            <Select
                options={mockList}
                callback={onClick}
            />
        );
        const selectButton = screen.getByTestId('select-button');
        const optionsList = screen.queryByTestId('dropdown-options');
        const option = screen.getByText(mockList[2].title);

        // клик по селекту и проверка видимости dropdown
        await userEvent.click(selectButton);
        expect(optionsList).toHaveClass('dropdown__options_active');

        // кликнули по выбранной опции, проверили закрытие dropdown
        await userEvent.click(option);
        expect(optionsList).not.toHaveClass('dropdown__options_active');

        // проверка изменения текста в селекте
        expect(selectButton).toContainHTML(mockList[2].title);
    });

    it('snapshot', () => {
        const select = render(<Select options={mockList} callback={onClick} />);
        expect(select).toMatchSnapshot()
    })
})