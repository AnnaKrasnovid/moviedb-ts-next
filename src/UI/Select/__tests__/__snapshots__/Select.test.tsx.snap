// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="dropdown"
      >
        <div
          class="dropdown__select "
        >
          <div
            class="select-button"
            data-testid="select-button"
          >
            <span
              class="
                select-button__title 
                select-button__title-button"
            >
              Все жанры
            </span>
            <span
              class="arrow "
              data-testid="arrow"
            >
              <svg
                class="arrow   "
                fill="none"
                height="22"
                viewBox="0 0 25 24"
                width="22"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M18.5 9L12.5 15L6.5 9"
                  stroke="#fff"
                  stroke-width="2"
                />
              </svg>
            </span>
          </div>
          <div
            class="
                    dropdown__options 
                    "
            data-testid="dropdown-options"
          >
            <div
              class="scroll"
            >
              <ul
                class="dropdown-select-list"
              >
                <li
                  class="
                        dropdown-select-list__item 
                        dropdown-select-list__item_active
                    "
                >
                  Все жанры
                </li>
                <li
                  class="
                        dropdown-select-list__item 
                        
                    "
                >
                  Боевик
                </li>
                <li
                  class="
                        dropdown-select-list__item 
                        
                    "
                >
                  Военный
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="dropdown"
    >
      <div
        class="dropdown__select "
      >
        <div
          class="select-button"
          data-testid="select-button"
        >
          <span
            class="
                select-button__title 
                select-button__title-button"
          >
            Все жанры
          </span>
          <span
            class="arrow "
            data-testid="arrow"
          >
            <svg
              class="arrow   "
              fill="none"
              height="22"
              viewBox="0 0 25 24"
              width="22"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18.5 9L12.5 15L6.5 9"
                stroke="#fff"
                stroke-width="2"
              />
            </svg>
          </span>
        </div>
        <div
          class="
                    dropdown__options 
                    "
          data-testid="dropdown-options"
        >
          <div
            class="scroll"
          >
            <ul
              class="dropdown-select-list"
            >
              <li
                class="
                        dropdown-select-list__item 
                        dropdown-select-list__item_active
                    "
              >
                Все жанры
              </li>
              <li
                class="
                        dropdown-select-list__item 
                        
                    "
              >
                Боевик
              </li>
              <li
                class="
                        dropdown-select-list__item 
                        
                    "
              >
                Военный
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
